cli/tests/integration/lsp_tests.rs

```diff
@@ -1545,35 +1545,25 @@ fn lsp_format_exclude_with_config() {
 #[test]
 fn lsp_format_exclude_default_config() {
   let temp_dir = TempDir::new().unwrap();
-  let workspace_root = temp_dir.path();
-
-  fs::write(
-    workspace_root.join("deno.json"),
-    br#"
-      {
-        "fmt": {
-          "files": {
-            "exclude": [ "ignored.ts" ]
-          }
-        }
-      }
-    "#,
-  )
-  .unwrap();
-
+  let workspace_root = temp_dir.path().canonicalize().unwrap();
```

Comment:
This was the fix, in that auto-discovery automatically canonicalizes the discovered path.

An open question though is how do clients behave?  Do they canonicalize files?  There isn't an easy way to determine.  We may need to fix `exclude` and `include` properties for things like format and lint to canoncilize files... 😕